### PR TITLE
Add --fail_on_assert and --no_store

### PR DESCRIPTION
**Pull request by martong**
Draft, to add extra options for better CI experience
* to fail the script when there is an assert in Clang
* to skip the store step.

In the CTU CI of E/// we don't really interested in storing the results. Also as the second step of a jenkins build pipeline we execute the analysis of those projects which must not assert at all (tmux, Xerces). However, if any of them fail we want to have an early exit with an error plus it is beneficial to see in the jenkins job which asserts happened.

### CODE REVIEW

**Comment by Xazax-hun** [comment]
I wonder if skipping stores is a good idea. Storing the result have still useful information, like tracking the number of defects, ability to diff ctu/non-ctu runs.
Some fields of the report is retrieved from the CodeChecker server.
Also, I think it would be better to have more fine grained control about fail on assertion. It could be a property of an analyzed project, so it could be set gradually for clean projects while we can still run on check on the failing projects without broking the CI. What do you think?

**Comment by martong** [comment]
>  fail on assertion. It could be a property of an analyzed project, so it could be set gradually for clean projects

Yes, this is a good idea. 

Also it would be nice to emit the results (possibly to stdout) immediately after we finished with one project. So we should not wait for the report until all projects are analized. But this is rather not a project property but a system wise setting.

> I wonder if skipping stores is a good idea. Storing the result have still useful information, like tracking the number of defects, ability to diff ctu/non-ctu runs.

Might be a bad idea, still it is not needed 99% of the time, so an option wouldn't harm anybody.

**Comment by Xazax-hun** [comment]
Does the store take significant amount of time compared to the rest of the pipeline? My concern is the following: maybe not having to store will save 5% of time. But once the results of the store is needed and you only realize once the job was already run, you need to rerun the whole job with this option disabled which is a 100%+ overhead. 

Having an option to emit the intermediate results is not a bad idea.

File: run_experiments.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import json
 import multiprocessing
 import os
+import pprint
```
Comment (Xazax-hun) [inline run_experiments.py]: What is this used for?

File: run_experiments.py [hunk 2]
```diff
@@ -463,7 +481,8 @@ def main():
         if not log_project(project, source_dir, args.jobs):
             continue
         check_project(project, source_dir, config, args.jobs)
-        post_process_project(project, source_dir, config, printer)
+        post_process_project(project, source_dir, config, printer,
```
Comment (Xazax-hun) [inline run_experiments.py]: I would prefer to make post_process return whether there were any errors or asserts and exit here based on `fail_on_assert`.  This way we could call printer.finish() before exiting and get a fully functional HTML report. We no longer would need to write the details to the stdout. Or if we still want to see the report in stdout as well, we could return the assert/error toplist from the post_process function.

**Comment by Xazax-hun** [comment]
I committed a similar solution.

**Comment by martong** [comment]
Thanks Gabor, sorry for my inactivity.

**Pull request closed without merging**